Initial import of perftools image sanity tests

happz opened the pull request:
Set of very simple tests, to verify basic usability of tools packaged in the image.

happz committed:
Initial import of perftools image sanity tests

Set of very simple tests, to verify basic usability of tools
packaged in the image.

happz committed:
Add short info about accepted params

hhorak commented on 4-perftools/test/expected-usage:
```diff
@@ -0,0 +1,21 @@
+
+and save that container to a reusable image:
+
+   % docker commit  devtoolset-4-perftools-rhel7-3.10.0-327.10.1.el7.x86_64
```
Thanks for submitting, I'm having only one issue -- this file hard-codes kernel build and so the test fails once the kernel does not match (which can happen quite easily). Would it be possible to rewrite the test the way that it does not compare this line?

happz replied:
Good catch, indeed the test fails on my Fedora laptop. Added a commit to address this issue, and tested on my laptop.

happz committed:
Make usage test independent on host kernel

"usage" script message contains output of `uname -r` which depends on
host kernel. Hardcoded release is deemed to fail whenever the test runs
in a different environment, e.g. on my Fedora laptop.

hhorak commented on 4-perftools/test/run:
```diff
@@ -44,6 +44,8 @@ function check_result() {
 function test_docker_run_usage () {
   info "Testing 'docker run' usage ..."
 
+  sed -i "s/RELEASE/$(docker run --rm $IMAGE_NAME uname -r)/" expected-usage
```
Hm, I don't think it is a good idea to change sources of cloned git repo. I'd rather suggest to store the sed output to `$TMPDIR/expected-usage` and compare that file with `$TMPDIR/actual-usage` instead.

happz replied:
Fair enough, I'll update the test.

happz replied:
Done.

hhorak commented on 4-perftools/test/run:
```diff
@@ -0,0 +1,114 @@
+  check_result "Exit code is zero" $? 1
+
+  grep "Your kernel's Performance Events Subsystem does not support your processor type." $TMPDIR/actual-ocount &> /dev/null
+  check_result "Output contains ocount's complaints" $? 0
```
This test `test_sanity_ocount_run` does not work for me, aren't there the exit codes swapped? `ocount ls` should exit with 0 and grep with 1, shouldn't it?

happz replied:
This is a weird one, as its result depends on host/guest/docker capabilities, whether perf events are available for dockerized tools - which was false in my testing environment (hence the non-zero exit code), apparently it's not the only option anymore (e.g. docker on my Fedora laptop works fine). Another thing is missing "--privileged" option. I guess the correct way forward would be to expect working perf, i.e. adjust the exit codes, and deal with the rest later.

happz replied:
Done.

hhorak commented on 4-perftools/test/run:
```diff
@@ -0,0 +1,114 @@
+  docker run --rm $IMAGE_NAME valgrind --version &> $TMPDIR/actual-valgrind-version
+  check_result "Exit code is zero" $? 0
+
+  grep 'valgrind-3.11.0' $TMPDIR/actual-valgrind-version &> /dev/null
```
This will work for RHEL image, but does not work in case of CentOS image, that still includes valgrind 3.10 (yeah, not deliberately, but that is what it is). I'm thinking whether the full version is important here, because that means we'll need to change the test once the valgrind is updated. Shouldn't it be enough to run `grep 'valgrind-3.1' $TMPDIR/actual-valgrind-version` as a check for the version?

happz replied:
Why does DTS image contain different version of valgrind in case of CentOS? I have to admit I never tried DTS on CentOS but I kind of thought that DTS as a SCL released by RedHat is somehow "fixed" in terms of package versions, until e.g. a bugfix is necessary, then erratum brings new DTS packages. Are there different DTS packages for CentOS?

hhorak replied:
There are some troubles with rebuilding the latest updates in cbs.centos.org, so it is just "delayed" a bit, but it may stay this way for some time, because there is noone working on it right now..

hhorak commented:
Few more nits, but it's almost ready.

happz committed:
Fixes sed and ocount usage

Addresses few notes by Honza Horak:

 - don't overwrite files in cloned repository - usage test now sed's
   into a temporary file, and uses that one in comparison

 - ocount test was tailored to fit a different environment, make it more
   public-ready by querying ophelp to find default event, and by
   expecting successfull run of ocount. Also, missing --privileged
   option could lead to weird behavior of this test.

hhorak commented:
Thanks for the changes, LGTM.

happz commented:
Awesome, thanks!
